curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/653639d63a45a077333312c8.md

```diff
@@ -7,23 +7,27 @@ dashedName: step-29
 
 # --description--
 
-Before playing the song, you need to make sure it starts from the beginning. This can be achieved by the use of the `currentTime` property of the `audio` object. 
+Before playing the song, you need to make sure it starts from the beginning. This can be achieved by the use of the `currentTime` property on the `audio` object. 
 
-Add an `if` statement to check whether the `userData?.currentSong` is `null` or if `userData?.currentSong.id` is strictly not equal `song.id`. Inside `if` block, set the `currentTime` property of the `audio` object to `0`.
+Add an `if` statement to check whether the `userData?.currentSong` is falsy *OR* if `userData?.currentSong.id` is strictly not equal to `song.id`. This condition will check if no current song is playing or if the current song is different from the one that is about to be played.
 
+Inside your `if` block, set the `audio.currentTime` property to `0`.
 
 # --hints--
 
-You should create an `if` statement with the condition `userData?.currentSong === null || userData?.currentSong.id !== song.id`.
+You should create an `if` statement that checks whether `userData.currentSong` is either falsy *OR* `userData.currentSong` is not equal to `song.id`. Don't forget to use optional chaining.
 
 ```js
-assert.match(code, /if\s*\(\s*userData\?\.currentSong\s*===\s*null\s*\|\|\s*userData\?\.currentSong\.id\s*!==\s*song\.id\s*\)\s*\{\s*/)
+assert.match(code, /if\s*\(\s*(?:!userData\?\.currentSong|\s*userData\?\.currentSong\s*===\s*null|\s*userData\?\.currentSong\s*===\s*undefined)\s*\|\|\s*userData\?\.currentSong\.id\s*!==\s*song\.id\s*|\s*userData\?\.currentSong\.id\s*!==\s*song\.id\s*\|\|\s*(?:!userData\?\.currentSong|\s*userData\?\.currentSong\s*===\s*null|\s*userData\?\.currentSong\s*===\s*undefined)\s*\)\s*\{\s*/);
```

Comment:
Since the first test checked for the condition and the various ways it can come in, I don't think we need to bring that in here again.

This looks tricky but it gets the job done. It checks for `audio.currentTime = 0;` inside curly braces.  

```suggestion
assert.match(code, /\{\s*audio\.currentTime\s*=\s*0;?\s*\}/);
```

Reply:
As much as this is shorter, unfortunately it doesn't check where exactly it was added.

Reply:
@Ksound22 We first need to account for all the possible code variations for `if` condition, only then we can check for `audio.currentTime = 0` inside of curly braces.